### DISCUSSION TITLE
Netbox LDAP issue

### DISCUSSION THREAD

**Discussion by szjanika** [netbox-community/netbox#7921] [General]
Recently upgraded the Netbox 2.9.x to 3.0.9. No issues. After the upgrade wanted to implement LDAP authentication. So I followed the Netbox documentation:
- Installed the required system packages
- Installed the django-auth-ldap package in the Python virtual environment.
- Modified the configureation.py (REMOTE_AUTH_BACKEND = 'netbox.authentication.LDAPBackend')
- Created ldap_config.py

After restarting netbox service I cannot login to the webgui.
Cannot authenticate with the already existing local admin account, too.
Neither any domain account.

So I turned on logging for authentication events.
This is the only error message I get:

initialize() argument 1 must be str, not set while authenticating user1

Same message on the gui after I hit login:

### <class 'TypeError'>

initialize() argument 1 must be str, not set

Python version: 3.9.2
NetBox version: 3.0.9

I was searching through the config files for "initialize()" without success...

I am not a Python guru, can somebody help me out?

**Comment by jeremystretch** [comment]
There's likely a syntax issue with your LDAP configuration file. Can you share your (sanitized) LDAP config? Also, if you can enable debug mode and share the complete stack trace, that will likely reveal the root cause.

**Comment by szjanika** [comment]
I found the typo in the LDAP config file and the LDAP authentication is working now.

Thank you for your help!
